### ISSUE TITLE
GetWallet missing balances

### ISSUE THREAD

**Issue by seertenedos** [enjin/enjin-csharp-sdk#56]
The result object for GetWallet does not have balances property.  I can't workout how to see the NFT's on an ethAddress the same as the graphql below returns in the playground.

query {
  GetWallet(ethAddress: "0x854Cd273c80eF85C3e674B0637AB067b962744F2") {
    ethAddress
    enjAllowance
    enjBalance
    ethBalance
    balances {
      asset{name}
      id
      index
      value
      project {
        name
      }
    }
    assetsCreated {
      id
      items {
        name
        id
      }
    }
  }
}

**Comment by seertenedos** [comment]
The same for getplayer. Really just trying to get all the info on a player when they login to make my app is insync with everything on the EthAddress of the user.

**Comment by ercancos** [comment]
Are there any progress? As far as I see from latest version of SDK, the problem still exist.

**Comment by BishopJohnson** [comment]
We'll be bringing support for this option in the next update for the SDK.

**Comment by ercancos** [comment]
I have taken your changes from development branch(by comparing your pull requests too.) and implemented on solution. Then I created new .dll for my project but as far as I see it still giving null when I try to get balances from getWallet result.
I tried both project client and player client versions of getWallet request. Both gave me the same result which is null.

Sample code that I used :
GetWallet req = new GetWallet();
        GraphqlResponse<Wallet> res = EnjinPlayerOperations.Instance.PlayerClientInstance.GetWallet(req).Result;

From that code I could successfully take eth. address of player but "res.Result.Balances" and all of the sub-domains gave me error since Balances is null.

Do you have any idea where did I wrong ? Or are the updated models not finished yet?

**Comment by BishopJohnson** [comment]
Hi @ercancos,

For certain requests, such as `GetWallet`, we use extension methods on the GraphQL fragment argument interfaces they implement as a way to offer options on what data may be returned for either conditionally nullable fields or fields that have a lot of data.

To request that the returned `Wallet` object have the list of balances it is necessary to use the `WithWalletBalances()` method on the request as so:
```
GetWallet req = new GetWallet().WithWalletBalances();
```

In addition, `GetPlayer` has the same extension methods as `GetWallet` with a few extra that are needed for player requests, so the following is now possible as well (assuming the player is linked):
```
GetPlayer req = new GetPlayer().WithWallet().WithWalletBalances();
```

I hope the above helps. Let us know if there's any other questions or concerns.

**Comment by ercancos** [comment]
> Hi @ercancos,
> 
> For certain requests, such as `GetWallet`, we use extension methods on the GraphQL fragment argument interfaces they implement as a way to offer options on what data may be returned for either conditionally nullable fields or fields that have a lot of data.
> 
> To request that the returned `Wallet` object have the list of balances it is necessary to use the `WithWalletBalances()` method on the request as so:
> 
> ```
> GetWallet req = new GetWallet().WithWalletBalances();
> ```
> 
> In addition, `GetPlayer` has the same extension methods as `GetWallet` with a few extra that are needed for player requests, so the following is now possible as well (assuming the player is linked):
> 
> ```
> GetPlayer req = new GetPlayer().WithWallet().WithWalletBalances();
> ```
> 
> I hope the above helps. Let us know if there's any other questions or concerns.

Thank you, It helped.

**Comment by BishopJohnson** [comment]
This issue ought to now be resolved in the latest pre-release, [1.0.0-alpha.6](https://github.com/enjin/enjin-csharp-sdk/releases/tag/1.0.0-alpha.6).
